test_responses.py

```diff
@@ -54,6 +56,34 @@ def run():
     assert_reset()
 
 
+def test_response_with_custom_reason():
+    @responses.activate
+    def run():
+        responses.add(
+            responses.GET,
+            "http://example.com",
+            body=b"take care!",
+            status=299,
+            reason="warning",
+        )
+        requests.get("http://example.com")
+        assert len(responses.calls) == 1
+        assert responses.calls[0].request.url == "http://example.com/"
+        assert responses.calls[0].response.content == b"take care!"
+        assert responses.calls[0].response.reason == "warning"
+        assert responses.calls[0].response.status_code == 299
+
+        requests.get("http://example.com?foo=bar")
```

Comment:
What is the purpose of this test?

Reply:
The test is similar to `test_response` just with custom status and reason. We can only test the newly added features. In my code I checked all values just to make an entire test.

Reply:
The test is similar to `test_response` just with custom status and reason. We can only test the newly added features. In my code I checked all values just to make an entire test.